bundles/org.openhab.binding.ws980wifi/src/main/java/org/openhab/binding/ws980wifi/internal/discovery/WS980WiFi.java

```diff
@@ -0,0 +1,286 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Joerg Dokupil - Initial contribution
+ */
+@NonNullByDefault
+public class WS980WiFi {
+
+    private Mac wsMac = new Mac("00:00:00:00:00:00");
+    private String wsHost = "";
+    private String wsDescription = "";
+    private @Nullable InetAddress wsIP = null;
+    private Integer wsPort = DATA_REQUEST_PORT; // port for TCP/IP requests e.g. for data extraction
+
+    public float tempInside; // in 0,1 Grad C (x10)
+    public float tempOutside; // in 0,1 Grad C (x10)
+    public float tempDewPoint; // in 0,1 Grad C (x10) Taupunkt
+    public float tempWindChill; // in 0,1 Grad C (x10) gefuehlteTemp
+    public float heatIndex; // in 0,1 Grad C (x10)
+    public float humidityInside; // in % innenFeuchte
+    public float humidityOutside; // in % aussenFeuchte
+    public float pressureAbsolut; // in 1/10 hPa (x10 = hPa) absDruck
+    public float pressureRelative; // in 1/10 hPa (x10 = hPa) relDruck
+    public float windDirection; // in Grad windRichtung
+    public float windSpeed; // in 1/10 m/s windGeschwindigkeit
+    public float windSpeedGust; // 12 Windboe
+    public float rainLastHour; // 0,1 mm regenStd
+    public float rainLastDay; // 0,1 mm regenTag
+    public float rainLastWeek; // 0,1 mm regenWoche
+    public float rainLastMonth; // 0,1 mm regenMonat
+    public float rainLastYear; // 0,1 mm regenJahr
+    public float rainTotal; // 0,1 mm regenGesamt
+    public float lightLevel; // 0,1 lux lichtStaerke
+    public float uvRaw; // uW/qm uvStaerke
+    public float uvIndex; // 1-12
+
+    protected final Logger log = LoggerFactory.getLogger(WS980WiFi.class);
+
+    public static final int DISCOVERY_DEST_PORT = 46000;
+    public static final int DATA_REQUEST_PORT = 45000;
+    public static final int DISCOVERY_RECEIVE_BUFFER_SIZE = 128;
+
+    protected WS980WiFi(Mac mac, InetAddress ip, String description, Integer port) {
+        this.wsMac = mac;
+        this.wsHost = ip.getHostName();
+        this.wsDescription = description;
+        this.wsIP = ip;
+        this.wsPort = port;
+    }
+
+    public WS980WiFi(String host, String port) {
+        this.wsHost = host;
+        this.wsPort = Integer.parseInt(port);
+    }
+
+    public String getMac() {
+        return wsMac.getMacString();
+    }
+
+    public String getIP() {
+        return wsIP.getHostAddress();
+    }
+
+    public String getHost() {
+        return wsHost;
+    }
+
+    public String getDescription() {
+        return wsDescription;
+    }
+
+    public Integer getPort() {
+        return wsPort;
+    }
+
+    public Boolean refreshValues() {
+        byte[] result = tcpipRequest(new byte[] { (byte) 0xff, (byte) 0xff, (byte) 0x0b, (byte) 0x00, (byte) 0x06,
+                (byte) 0x04, (byte) 0x04, (byte) 0x19 });
+        if (result.length > 0) {
+            this.tempInside = (float) (Long.decode("0x" + bytesToHex(Arrays.copyOfRange(result, 7, 9)))) / 10;
```

Comment:
Validation needed, `ArrayIndexOutOfBoundsException` possible since `result.length` is unchecked.